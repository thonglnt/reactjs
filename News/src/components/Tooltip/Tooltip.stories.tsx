// lib
import { Story } from '@storybook/react'

// Components
import Tooltip, { TooltipProps } from '.'

export default {
  title: 'Components/Tooltip',
  component: Tooltip,
}

const Template: Story<TooltipProps> = (args) => (
  <div style={{ backgroundColor: 'black', width: '100px', height: '100px' }}>
    <Tooltip {...args} />
  </div>
)

export const TooltipBusiness = Template.bind({})
TooltipBusiness.args = {
  nameTag: 'Business',
}

export const TooltipEntertainment = Template.bind({})
TooltipEntertainment.args = {
  nameTag: 'Entertainment',
}
