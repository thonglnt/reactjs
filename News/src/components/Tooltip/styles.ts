// lib
import styled from 'styled-components'

// themes
import { theme } from '@themes'

export const StyledToolTip = styled.a`
  border: 1px solid ${theme.Colors.white};
  padding: 0 4px;
  text-decoration: none;
  color: ${theme.Colors.white};

  &:hover {
    background-color: ${theme.Colors.crayola};
    border: none;
    transition: all 0.3s;
  }
`
