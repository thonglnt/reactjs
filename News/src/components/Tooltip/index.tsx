// lib
import { memo } from 'react'

// style
import { StyledToolTip } from './styles'

export type TooltipProps = {
  nameTag: string
  link?: string
}

const Tooltip = ({ nameTag = '', link = '' }: TooltipProps) => {
  return (
    <div>
      <StyledToolTip href={link}>{nameTag}</StyledToolTip>
    </div>
  )
}

export default memo(Tooltip)
