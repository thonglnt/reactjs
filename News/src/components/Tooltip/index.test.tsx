// lib
import { render } from '@testing-library/react'

// components
import Tooltip, { TooltipProps } from '.'

describe('Test Tooltip component', () => {
  const defaultProps: TooltipProps = {
    nameTag: 'Business',
  }

  test('It should match DOM Snapshot', () => {
    const { container } = render(<Tooltip {...defaultProps} />)

    expect(container).toMatchSnapshot()
  })

  test('Component Tooltip should render correctly', () => {
    const { getByText } = render(<Tooltip {...defaultProps} />)

    expect(getByText('Business')).toBeInTheDocument()
  })
})
