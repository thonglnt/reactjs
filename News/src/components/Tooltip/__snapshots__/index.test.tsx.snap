// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component It should match DOM Snapshot 1`] = `
<div>
  <div>
    <a
      class="sc-bcXHqe dbEkjq"
      href=""
    >
      Business
    </a>
  </div>
</div>
`;
